### PR TITLE
Feature/blui 2894 lic crawler

### PR DESCRIPTION
**Pull request by jeffvg**
Fixes # .

#### Changes proposed in this Pull Request:
- add npm-license-crawler & script
- add license.json
- 

#### To Test:
-

### CODE REVIEW

**Comment by huayunh** [comment]
Want to set up a license crawler for `figma-plugins/_maintenance_/icon-description/` too?

**Comment by jeffvg** [comment]
> Want to set up a license crawler for `figma-plugins/_maintenance_/icon-description/` too?

@huayunh add the lic crawler for icon description

File: variant-switcher/package.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
         "prettier": "prettier \"src/**/**.{ts,tsx,js,jsx,json,css,scss,html,md}\" --write",
         "test": "echo \"Error: no test specified\" && exit 1",
         "build": "npx webpack",
-        "build:production": "npx webpack --mode production"
+        "build:production": "npx webpack --mode production",
+        "generate:licenses": "npm-license-crawler -onlyDirectDependencies -json LICENSES.json && cd .. && cd ./_maintenance_/icon-description && yarn generate:licenses"
```
Comment (huayunh) [inline variant-switcher/package.json:11]: Variant Switcher and Icon Description are two separate plugins. I do not expect licenses in icon-description getting updated when I run yarn generate:licenses in Variant Switcher's folder
```suggestion
        "generate:licenses": "npm-license-crawler -onlyDirectDependencies -json LICENSES.json"
```

**Review by huayunh — COMMENTED** [review]
When I ran `yarn generate:licenses`, `yarn.lock` files in  both variant-switcher and icon-description are changed
image

File: variant-switcher/package.json (hunk 1 above)
Comment (jeffvg) [reply variant-switcher/package.json:11]: not sure I understand, what version of node are you running? 

we can change the script in variant switcher to only generate a license for variant switcher. You'll just have to remember if you add another dependency in icon description that you'll need to run generate licenses there

**Comment by jeffvg** [comment]
I'm not sure why your lock file updates and removes the (#) and is replaced with (.git)
From what I see in other PR's the lock file updates is using (#)

**Review by huayunh — APPROVED** [review]

**Pull request merged**